pyppeteer/target.py

```diff
@@ -52,24 +52,36 @@ async def createCDPSession(self) -> CDPSession:
         """Create a Chrome Devtools Protocol session attached to the target."""
         return await self._sessionFactory()
 
+    async def _create_page(self) -> Page:
+        """Create the page of this target."""
+        client = await self._sessionFactory()
+        new_page = await Page.create(
+            client, self,
+            self._ignoreHTTPSErrors,
+            self._defaultViewport,
+            self._screenshotTaskQueue,
+        )
+        return new_page
+
     async def page(self) -> Optional[Page]:
         """Get page of this target.
 
         If the target is not of type "page" or "background_page", return
         ``None``.
         """
-        if (self._targetInfo['type'] in ['page', 'background_page'] and
-                self._page is None):
-            client = await self._sessionFactory()
-            new_page = await Page.create(
-                client, self,
-                self._ignoreHTTPSErrors,
-                self._defaultViewport,
-                self._screenshotTaskQueue,
-            )
-            self._page = new_page
-            return new_page
-        return self._page
+        if self._page is None:
+            if self._targetInfo['type'] in ['page', 'background_page']:
+                # Creation of page is seperated as a task for immediate updation
+                # of the self._page variable
+                self._page = new_page = asyncio.create_task(self._create_page())
+                return await new_page
```

Comment:
Sorry the intent of this doesn't seem entirely clear to me, is there any difference in the following code? 
```
                return await self._create_page()
```

Reply:
On usage i happen to find that this particular function gets called _twice_ almost at the same time. This leads to creation of 2 Page objects for the same tab/page. One of the Page objects gets added to the pages list, and one is just dropped off. By storing the page creation function as an Task object in the instance variable self._page., the first call works as before. Now, the second time function is called, the function doesnt create another Page object but instead awaits from the instance variable self._page, and hence both the calls would return the **same Page object**. (please check line 83, 84). 

I just notice that just viewing the highlighted changes, my line seems unnecessary; I think you would understand better if you looked at the complete function in the file.

Reply:
I realize taken in isolation, yeah it looks unecessary, but I did read through the whole changeset. I'll take a look at the relevant bug and try to reproduce here so that I know exactly what's going on. Thanks for sending in this PR BTW!